prow/cmd/branchprotector/BUILD

```diff
@@ -1,3 +1,44 @@
+# Usage:
+#   bazel run :dev-job.create
+load("@io_bazel_rules_k8s//k8s:object.bzl", "k8s_object")
+load("@io_bazel_rules_docker//docker:docker.bzl", "docker_bundle")
+load("@io_bazel_rules_docker//contrib:push-all.bzl", "docker_push")
+load("@io_bazel_rules_docker//go:image.bzl", "go_image")
+
+k8s_object(
+    name = "dev-job",
+    cluster = "gke_k8s-prow_us-central1-f_prow",  # Run make :get-cluster-credentials
+    images = {
+        # TODO(fejta): ideally {STABLE_DOCKER_REPO}
+        # Needs to match the value in dev-job.yaml
+        "gcr.io/k8s-testimages/branchprotector:latest": ":branchprotector-image",
+    },
+    kind = "Job",
+    template = ":dev-job.yaml",
+)
+
+docker_bundle(
+    name = "bundle",
+    images = {
+        "{STABLE_PROW_REPO}/branchprotector:{DOCKER_TAG}": ":branchprotector-image",
+        "{STABLE_PROW_REPO}/branchprotector:latest": ":branchprotector-image",
+        "{STABLE_PROW_REPO}/branchprotector:latest-{BUILD_USER}": ":branchprotector-image",
+    },
+    stamp = True,
+)
+
+docker_push(
+    name = "push",
+    bundle = ":bundle",
+)
+
+go_image(
```

Comment:
If there is a `go_binary` target in this `BUILD` file, you can now just say `binary = ":that"` here to wrap it in an image.  This is available for the `lang_image` rules that don't benefit from clever layering.

I did this as a workaround/compromise since Gazelle will only manage `go_binary` and not `go_image`.